src/copilot/tools/listSchemasTool.ts

```diff
@@ -0,0 +1,113 @@
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import { ToolBase } from "./toolBase";
+import ConnectionManager from "../../controllers/connectionManager";
+import * as Constants from "../../constants/constants";
+import { MssqlChatAgent as loc } from "../../constants/locConstants";
+import { getErrorMessage } from "../../utils/utils";
+import SqlToolsServiceClient from "../../languageservice/serviceclient";
+import { RequestType } from "vscode-languageclient";
+import { SimpleExecuteResult } from "../interfaces";
+import { listSchemasQuery } from "../queries";
+
+export interface ListSchemasToolParams {
+    connectionId: string;
+}
+
+export interface ListSchemasToolResult {
+    success: boolean;
+    message?: string;
+    schemas: string[];
+}
+
+export class ListSchemasTool extends ToolBase<ListSchemasToolParams> {
+    public readonly toolName = Constants.copilotListSchemasToolName;
+
+    constructor(
+        private _connectionManager: ConnectionManager,
+        private _client: SqlToolsServiceClient,
+    ) {
+        super();
+    }
+
+    async call(
+        options: vscode.LanguageModelToolInvocationOptions<ListSchemasToolParams>,
+        _token: vscode.CancellationToken,
+    ) {
+        const { connectionId } = options.input;
+        try {
+            const connInfo = this._connectionManager.getConnectionInfo(connectionId);
+            const connCreds = connInfo?.credentials;
+            if (!connCreds) {
+                return JSON.stringify({
+                    success: false,
+                    message: loc.noConnectionError(connectionId),
+                });
+            }
+
+            const result = await this._client.sendRequest(
+                new RequestType<
+                    { ownerUri: string; queryString: string },
+                    SimpleExecuteResult,
+                    void,
+                    void
+                >("query/simpleexecute"),
+                {
+                    ownerUri: connectionId,
+                    queryString: listSchemasQuery,
+                },
+            );
+            const schemas = this.getSchemaNamesFromResult(result);
+
+            return JSON.stringify({
+                success: true,
+                schemas,
+            });
+        } catch (err) {
+            return JSON.stringify({
+                success: false,
+                message: getErrorMessage(err),
+            });
+        }
+    }
+
+    private getSchemaNamesFromResult(result: SimpleExecuteResult): string[] {
+        if (!result || !result.rows || result.rows.length === 0) {
+            return [];
+        }
+
+        const schemaNames: string[] = [];
+
+        // Extract schema names from each row
+        // Assuming the query returns schema names in the first column
```

Comment:
Add a similar comment to this one in case the query is modified for some reason and causes schema names to appear in a different column that isn't first.

Reply:
Added a warning section with the assumption to the top of `queries.ts`